heuristic/classes/Problem.py

```diff
@@ -85,6 +85,13 @@ def smallest_quantity_customers(self) -> np.ndarray:
         """
         return np.argsort(self._demands + self._pickups)
 
+    @classmethod
+    def clear(cls):
```

Comment:
Wrong place. Move this up into Singleton, and have it clear the instances cache. I'm not sure what you're doing right now, but this is.. weird.